src/redux/rockets/RocketsSlice.js

```diff
@@ -0,0 +1,62 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const api = 'https://api.spacexdata.com/v4/rockets';
+
+const initialState = {
+  rocketList: [],
+  status: 'idle',
+  error: null,
+};
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', async () => {
+  try {
+    const response = await axios.get(api);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const rocketsSlice = createSlice({
+  name: 'rockets',
+  initialState,
+  reducers: {
+    reserveRocket: (state, payload) => ({
+      ...state,
+      rocketList: state.rocketList.map((rocket) => {
+        if (rocket.id === payload.payload) {
+          if (rocket.reserved === true) {
+            return { ...rocket, reserved: false };
+          }
+          return { ...rocket, reserved: true };
+        }
+        return rocket;
+      }),
+    }),
+  },
```

Comment:
- Optional
I strongly recommend having a separate reducer for canceling the rocket reservation. It is always recommended to have one reducer per action.